### PR TITLE
Randomized Round-Start Names for Voidcrafts

### PR DESCRIPTION
**Pull request by Bobbanz1**
## About The Pull Request
This pull request makes it so that any voidcraft on the map when the map is loaded gets assigned a randomized name in the format of: [Noun] of [Celestial Body]. 

The code has three lists: 
One list contains nouns. 
Another contains words for planets and gas giants, taken from Wikipedia, some of these can be found in our solar system, others are exoplanets. 
The third list contains names of moons found in our solar system. 

The code picks one noun and then generates a number between 0 and 1 to decide between true or false. 
If it’s true, then it picks a name from the Moon List. 
If it’s false then it picks from the planet list.

A few examples of possible combinations and impossible combinations are: 
Chicken of Terra
Pounder of Uranus
Paw of Uranus
Eye of Prometheus
Wings of Luna.

## Why It's Good For The Game
De-Hegemonizes the voidcrafts initial names from Number-Raptor/Scimitar/Rapier, to names/callsigns that give them a slight starting sense of individuality, which pilots usually do during a round.

Names have sometimes been assigned to aircrafts, one example being bombers from WW2, two such examples being the “Ye Olde Pub” and “All American”

## Changelog
:cl:
add: Added three lists and a name generator which picks from two of these three lists and slaps them together into a single name. 
add: Added a line of code to the initialization of fighters to use the name generator and give themselves a random name
del: Removed Uranus from the List, it ruins everything.
/:cl:

### CODE REVIEW

**Review by TheFakeElon — COMMENTED** [review]
put this in a txt file or JSON and load it from there

File: nsv13/code/__HELPERS/names.dm [hunk 1]
```diff
@@ -38,3 +38,66 @@ var/list/ship_name_suffix = list("slayer","blazer","bringer","slapper","twirler"
...
+
+/var/list/fighter_noun = list(
+	"Paw","Hand","Warmth","Eye","Wings","Palms","Pillar",
+	"Dawn","Fist","Chicken","Hawk","Sparrow","Thunder","Watcher",
+	"Guardian","Stalker","Heron","Swan","Swallow","The Unladen Swallow",
+	"Wren","Corvid","Falcon","Eagles","Owl","Kestrel","Harrier","Vultures",
+	"Osprey","Parrot","Kingfisher","Toucan","Trogon","Inflation","Union",
+	"Cayenne","Mortgage","Hen","Light","Puma","Aglet","Seller","Steam",
+	"Lighting","Wriggler","Carp","Ginseng","Hyena","Tenement","Jet","Cheese",
+	"Asterisk")
```
Comment (TheFakeElon) [inline nsv13/code/__HELPERS/names.dm:50]: This is not how you are supposed to do global variables, very bad. 🗞️
you'd use the GLOBAL_LIST_INIT define

Comment (Bobbanz1) [reply nsv13/code/__HELPERS/names.dm]: Done

**Comment by Bobbanz1** [comment]
I’m sorry if it’s sent the request for a re-review multiple times as the ui doesn’t seem that intuitive in terms of notifying the user that they have indeed sent the request or not.

**Review by TheFakeElon — COMMENTED** [review]
Looks much better :)

Though it's generally a good idea to put global definitions in the `_globalvars` folder, it makes keeping track of them much easier and can prevent override issues. Something like `nsv13/code/_globalvars/lists/names.dm` would probably work well!

File: nsv13/code/__HELPERS/names.dm (hunk 1 above)
Comment (Bokkiewokkie) [reply nsv13/code/__HELPERS/names.dm]: > This is not how you are supposed to do global variables, very bad. 🗞️ you'd use the GLOBAL_LIST_INIT define

Yes instead we use var/static/list inside the proc that generates the names :)

File: nsv13/code/__HELPERS/names.dm [hunk 2]
```diff
@@ -38,3 +38,14 @@ var/list/ship_name_suffix = list("slayer","blazer","bringer","slapper","twirler"
...
 	return capitalize(name)
+
+/proc/generate_fighter_name()
+	var/name = ""
+	var/randomizer = rand(0,1)
+	var/chooser = FALSE
+	if(randomizer == 0)
+		chooser = FALSE
+	else
+		chooser = TRUE
```
Comment (Bokkiewokkie) [inline nsv13/code/__HELPERS/names.dm:49]: 1 and 0 are already TRUE and FALSE in byond, making randomizer and chooser the same (if one is 1 it is also TRUE and vice versa). The next comment contains the preferred solution.

File: nsv13/code/__HELPERS/names.dm [hunk 3]
```diff
@@ -38,3 +38,14 @@ var/list/ship_name_suffix = list("slayer","blazer","bringer","slapper","twirler"
...
+	name = pick(GLOB.fighter_noun) + " of " + "[chooser ? pick(GLOB.fighter_moon) : pick(GLOB.fighter_planet)]"
+	return capitalize(name)
```
Comment (Bokkiewokkie) [inline nsv13/code/__HELPERS/names.dm:51]:
```suggestion
	name = pick(GLOB.fighter_noun) + " of " + "[pick(0,1) ? pick(GLOB.fighter_moon) : pick(GLOB.fighter_planet)]"

```
Just put the randomization in here and it'll work perfectly fine, no need to overcomplicate things.
Your text is all already capitalized in the file, also don't forget to put empty lines at the end of your files

File: nsv13/code/_globalvars/lists/names.dm [hunk 4]
```diff
@@ -0,0 +1,3 @@
+GLOBAL_LIST_INIT(fighter_noun, world.file2list("strings/names/fighter_noun.txt"))
+GLOBAL_LIST_INIT(fighter_planet, world.file2list("strings/names/fighter_planet.txt"))
+GLOBAL_LIST_INIT(fighter_moon, world.file2list("strings/names/fighter_moon.txt"))
```
Comment (Bokkiewokkie) [inline nsv13/code/_globalvars/lists/names.dm:3]:
```suggestion
GLOBAL_LIST_INIT(fighter_noun, world.file2list("strings/names/fighter_noun.txt"))
GLOBAL_LIST_INIT(fighter_planet, world.file2list("strings/names/fighter_planet.txt"))
GLOBAL_LIST_INIT(fighter_moon, world.file2list("strings/names/fighter_moon.txt"))

```
Ditto

**Review by Bokkiewokkie — CHANGES_REQUESTED** [review]

**Comment by Bobbanz1** [comment]
There’s a grand total of 16 848 possible name combinations if I did the mathematics correctly.

**Review by covertcorvid — APPROVED** [review]

**Comment by Karmic-Skink** [comment]
image

Typo

**Review by Bokkiewokkie — APPROVED** [review]

**Pull request merged**